development/openstack-base-focal-wallaby/README.md

```diff
@@ -1,269 +1,371 @@
 
-    openstack subnet create --network internal \
-        --subnet-range 198.51.100.0/24 \
-        --dns-nameserver 8.8.8.8 \
-        internal_subnet
+    curl http://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img
+       --output ~/cloud-images/focal-amd64.img
 
-    openstack router create provider-router
+Now import the image and call it 'focal-amd64':
+
+    openstack image create --public --container-format bare \
+       --disk-format qcow2 --file ~/cloud-images/focal-amd64.img \
+       focal-amd64
+
+Images for other Ubuntu releases and architectures can be obtained in a similar
+way.
+
+For the ARM 64-bit (arm64) architecture you will need to configure the image to
+boot in UEFI mode:
+
+    curl http://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img
+       --output ~/cloud-images/focal-arm64.img
+
+    openstack image create --public --container-format bare \
+       --disk-format qcow2 --property hw_firmware_type=uefi \
+       --file ~/cloud-images/focal-arm64.img \
+       focal-arm64
+
+## Configure networking
+
+For the purposes of a quick test, we'll set up an external network and a
+shared router ('provider-router') that will be used by all tenants for public
+access to instances.
 
+For an example private cloud, create a network ('ext_net'):
+
+    openstack network create --external \
+       --provider-network-type flat --provider-physical-network physnet1 \
+       ext_net
+
+When creating the external subnet ('ext_subnet') the actual values used will
+depend on the environment that the second network interface (on all nodes) is
+connected to:
+
+    openstack subnet create --network ext_net --no-dhcp \
+       --gateway 10.0.0.1 --subnet-range 10.0.0.0/21 \
+       --allocation-pool start=10.0.0.10,end=10.0.0.200 \
+       ext_subnet
+
+> **Note**: For a public cloud the ports would be connected to a publicly
+  addressable part of the internet.
+
+We'll also need an internal network ('int_net'), subnet ('int_subnet'), and
+router ('provider-router'):
+
+    openstack network create int_net
+
+    openstack subnet create --network int_net --dns-nameserver 8.8.8.8 \
+       --gateway 192.168.0.1 --subnet-range 192.168.0/24 \
+       --allocation-pool start=192.168.0.10,end=192.168.0.200 \
+       int_subnet
+
+    openstack router create provider-router
     openstack router set --external-gateway ext_net provider-router
+    openstack router add subnet provider-router int_subnet
+
+See the [Neutron documentation][openstack-neutron] for more information.
+
+## Create a flavor
 
-    openstack router add subnet provider-router internal_subnet
+Create at least one flavor to define a hardware profile for new instances. Here
+we create one called 'm1.tiny':
 
-Neutron provides a wide range of configuration options; see the [OpenStack Neutron][] documentation for more details.
+    openstack flavor create --ram 1024 --disk 10 --ephemeral 10 m1.tiny
```

Comment:
The reason we have used m1.small in the past is some Ubuntu images will not fit on an m1.tiny. I suspect we want to keep those settings.

Reply:
Ok